modules/social_features/social_embed/src/Controller/EmbedController.php

```diff
@@ -0,0 +1,142 @@
+   *
+   * @var \Drupal\Core\Flood\FloodInterface
+   */
+  protected FloodInterface $flood;
+
+  /**
+   * The EmbedController constructor.
+   *
+   * @param \Drupal\url_embed\UrlEmbed $url_embed
+   *   The url embed services.
+   * @param \Drupal\Core\Flood\FloodInterface $flood
+   *   The flood service.
+   */
+  public function __construct(UrlEmbed $url_embed, FloodInterface $flood) {
+    $this->urlEmbed = $url_embed;
+    $this->flood = $flood;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('url_embed'),
+      $container->get('flood')
+    );
+  }
+
+  /**
+   * Generates embed content of a give URL.
+   *
+   * When the site-wide setting for consent is enabled, the links in posts and
+   * nodes will be replaced with placeholder divs and a show content button.
+   *
+   * Once user clicks the button, it will send request to this controller along
+   * with url of the content to embed and an uuid which differentiates each
+   * link.
+   *
+   * See:
+   * 1. SocialEmbedConvertUrlToEmbedFilter::convertUrls
+   * 2. SocialEmbedUrlEmbedFilter::process
+   * 3. EmbedConsentForm::buildForm
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The current request object.
+   *
+   * @return \Drupal\Core\Ajax\AjaxResponse
+   *   The Ajax response.
+   */
+  public function generateEmbed(Request $request) {
+    // Get the requested URL of content to embed.
+    $url = $request->query->get('url');
+    // Get unique identifier for the button which was clicked.
+    $uuid = $request->query->get('uuid');
+
+    // If $url or $uuid is not present, then request is malformed.
+    if ($url == NULL && Uuid::isValid($uuid) != FALSE) {
+      throw new NotFoundHttpException();
+    }
+
+    // Let's prepare a unique flood identifier using our uuid.
+    // This is to to ensure that the value doesn't exceed 128 characters.
+    // As identifiers cannot be more than 128 characters long, otherwise
+    // it will cause an error when attempting to write the flood record
+    // to the database.
+    $hashedValue = Crypt::hashBase64($uuid);
+    // The reason we are also using the combination of uuid and client IP is
+    // to make sure we do not end up blocking a user who is just exploring our
+    // website. There can be a scenario where, this endpoint can be called more
+    // than our threshold value due to numerous embedded content on the
+    // platform, and if we use only clientIp, we might end up blocking a
+    // genuine request.
+    $floodIdentifier = $request->getClientIP() . $hashedValue;
+
+    // Only proceed if this is not a malicous request.
+    if (!$this->flood->isAllowed('social_embed.generateembed_flood_event', 50, 3600, $floodIdentifier)) {
```

Comment:
I think we need this to be configurable, and definitely Unit tested.
It seems like in this case a user passing on a different $uuid can still DDOS the system because the floodIdentifier, which gives a user 50 retries consists of the IP + the UUID. So by changing the UUID, you'll get 50 retries.
I know we need to worry about a stream with mulitple embeds, but even than a user will never have to hit that limit if used correctly.